--- ./CMake/FindMesaSrc.cmake~	2014-04-17 20:18:21.000000000 +0100
+++ ./CMake/FindMesaSrc.cmake	2014-04-17 20:30:40.120637372 +0100
@@ -18,7 +18,6 @@
                          ${MESA_SOURCE_PREFIX}/include
                          ${MESA_SOURCE_PREFIX}/src/mapi
                          ${MESA_SOURCE_PREFIX}/src/mesa/drivers/dri/i965/
-                         ${MESA_SOURCE_PREFIX}/src/mesa/drivers/dri/intel/
                          ${MESA_SOURCE_PREFIX}/src/mesa/drivers/dri/common/)
 SET(MESA_SOURCE_FOUND 1 CACHE STRING "Set to 1 if mesa source code is found, 0 otherwise")
 ELSE(MESA_SOURCE_PREFIX)
