--- ./backend/src/CMakeLists.txt~	2015-12-16 13:53:19.000000000 +0100
+++ ./backend/src/CMakeLists.txt	2015-12-16 14:21:03.461805319 +0100
@@ -115,8 +115,6 @@
     backend/gen75_context.cpp
     backend/gen8_context.hpp
     backend/gen8_context.cpp
-    backend/gen9_context.hpp
-    backend/gen9_context.cpp
     backend/gen_program.cpp
     backend/gen_program.hpp
     backend/gen_program.h
@@ -132,8 +130,6 @@
     backend/gen75_encoder.cpp
     backend/gen8_encoder.hpp
     backend/gen8_encoder.cpp
-    backend/gen9_encoder.hpp
-    backend/gen9_encoder.cpp
     )
 
 set (GBE_LINK_LIBRARIES
