diff --git a/backend/src/llvm/llvm_scalarize.cpp b/backend/src/llvm/llvm_scalarize.cpp
index b48ff81..89121ea 100644
--- a/backend/src/llvm/llvm_scalarize.cpp
+++ b/backend/src/llvm/llvm_scalarize.cpp
@@ -741,7 +741,7 @@ namespace gbe {
             Value* foo_i = getComponent(i, foo);
             assert(foo_i && "There is unhandled vector component");
             Value* idxs_i[] = {ConstantInt::get(intTy,0), ConstantInt::get(intTy,i)};
-            Instruction* storePtr_i = GetElementPtrInst::Create(Alloc, idxs_i, "", extr);
+            Instruction* storePtr_i = GetElementPtrInst::Create(fooTy, Alloc, idxs_i, "", extr);
             new StoreInst(foo_i, storePtr_i, extr);
           }
           vectorAlloca[foo] = Alloc;
@@ -749,7 +749,7 @@ namespace gbe {
         else Alloc = dyn_cast<Instruction>(vectorAlloca[foo]);
 
         Value* Idxs[] = {ConstantInt::get(intTy,0), extr->getOperand(1)};
-        Instruction* getPtr = GetElementPtrInst::Create(Alloc, Idxs, "", extr);
+        Instruction* getPtr = GetElementPtrInst::Create(fooTy, Alloc, Idxs, "", extr);
         Instruction* loadComp = new LoadInst(getPtr,"",extr);
         extr->replaceAllUsesWith(loadComp);
         return true;
