--- ./backend/src/llvm/llvm_gen_backend.cpp~	2013-12-17 06:39:10.000000000 +0000
+++ ./backend/src/llvm/llvm_gen_backend.cpp	2013-12-17 06:42:15.736723355 +0000
@@ -154,9 +154,9 @@
 #define LLVM_VERSION_MINOR 0
 #endif /* !defined(LLVM_VERSION_MINOR) */
 
-#if (LLVM_VERSION_MAJOR != 3) || (LLVM_VERSION_MINOR > 4)
-#error "Only LLVM 3.0 - 3.4 is supported"
-#endif /* (LLVM_VERSION_MAJOR != 3) || (LLVM_VERSION_MINOR > 4) */
+#if (LLVM_VERSION_MAJOR != 3) || (LLVM_VERSION_MINOR > 5)
+#error "Only LLVM 3.0 - 3.5 is supported"
+#endif /* (LLVM_VERSION_MAJOR != 3) || (LLVM_VERSION_MINOR > 5) */
 
 using namespace llvm;
 
