--- ./benchmark/CMakeLists.txt~	2014-04-30 18:19:34.000000000 +0100
+++ ./benchmark/CMakeLists.txt	2014-04-30 18:26:23.314810046 +0100
@@ -14,8 +14,8 @@
 
 ADD_LIBRARY(benchmarks SHARED ${ADDMATHFUNC} ${benchmark_sources})
 
-#TARGET_LINK_LIBRARIES(benchmarks cl m ${OPENGL_LIBRARIES} ${CMAKE_THREAD_LIBS_INIT})
-TARGET_LINK_LIBRARIES(benchmarks cl m)
+#TARGET_LINK_LIBRARIES(benchmarks OpenCL m ${OPENGL_LIBRARIES} ${CMAKE_THREAD_LIBS_INIT})
+TARGET_LINK_LIBRARIES(benchmarks OpenCL m)
 
 ADD_EXECUTABLE(benchmark_run benchmark_run.cpp)
 TARGET_LINK_LIBRARIES(benchmark_run benchmarks)
--- ./src/CMakeLists.txt.orig	2014-11-05 16:19:18.874552283 +0000
+++ ./src/CMakeLists.txt	2014-11-05 18:37:01.867901298 +0000
@@ -118,10 +118,11 @@
 SET(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wl,-Bsymbolic,--allow-shlib-undefined")
 
 link_directories (${LLVM_LIBRARY_DIR} ${DRM_LIBDIR})
-add_library(cl SHARED ${OPENCL_SRC})
-ADD_DEPENDENCIES(cl ${GIT_SHA1})
+add_library(OpenCL SHARED ${OPENCL_SRC})
+set_target_properties(OpenCL PROPERTIES VERSION 1.0.0 SOVERSION 1)
+ADD_DEPENDENCIES(OpenCL ${GIT_SHA1})
 target_link_libraries(
-                      cl
+                      OpenCL
                       ${X11_LIBRARIES}
                       ${XEXT_LIBRARIES}
                       ${XFIXES_LIBRARIES}
@@ -131,4 +132,4 @@
                       ${CMAKE_DL_LIBS}
                       ${OPENGL_LIBRARIES}
                       ${OPTIONAL_EGL_LIBRARY})
-install (TARGETS cl LIBRARY DESTINATION ${BEIGNET_INSTALL_DIR})
+install (TARGETS OpenCL LIBRARY DESTINATION ${BEIGNET_INSTALL_DIR})
--- ./utests/CMakeLists.txt.orig	2014-11-27 21:49:11.591745360 +0000
+++ ./utests/CMakeLists.txt	2014-11-27 21:50:31.355739020 +0000
@@ -257,7 +257,7 @@
 
 ADD_LIBRARY(utests SHARED ${ADDMATHFUNC} ${utests_sources})
 
-TARGET_LINK_LIBRARIES(utests cl m ${OPENGL_LIBRARIES} ${UTESTS_REQUIRED_EGL_LIB} ${CMAKE_THREAD_LIBS_INIT} ${UTESTS_REQUIRED_X11_LIB})
+TARGET_LINK_LIBRARIES(utests OpenCL m ${OPENGL_LIBRARIES} ${UTESTS_REQUIRED_EGL_LIB} ${CMAKE_THREAD_LIBS_INIT} ${UTESTS_REQUIRED_X11_LIB})
 
 ADD_EXECUTABLE(utest_run utest_run.cpp)
 TARGET_LINK_LIBRARIES(utest_run utests)
