--- ./backend/src/llvm/llvm_bitcode_link.cpp~	2015-02-12 12:48:52.000000000 +0000
+++ ./backend/src/llvm/llvm_bitcode_link.cpp	2015-02-12 13:04:42.627791705 +0000
@@ -224,7 +224,7 @@
       }
 #else
         if (std::error_code EC = newMF->materialize()) {
-          printf("Can not materialize the function: %s, because %s\n", fnName.c_str(), EC.message().c_str();
+          printf("Can not materialize the function: %s, because %s\n", fnName.c_str(), EC.message().c_str());
           delete clonedLib;
           return NULL;
         }
