--- ./src/mesa/swrast/s_aatriangle.c~	2016-10-25 16:57:09.000000000 +0100
+++ ./src/mesa/swrast/s_aatriangle.c	2016-10-25 17:19:03.305682708 +0100
@@ -28,6 +28,7 @@
  */
 
 
+#include <omp.h>
 #include "main/glheader.h"
 #include "main/context.h"
 #include "main/macros.h"
--- ./src/mesa/swrast/s_context.c~	2016-10-25 16:57:09.000000000 +0100
+++ ./src/mesa/swrast/s_context.c	2016-10-25 17:19:41.843088794 +0100
@@ -25,6 +25,7 @@
  *    Keith Whitwell <keithw@vmware.com> Brian Paul
  */
 
+#include <omp.h>
 #include "main/imports.h"
 #include "main/bufferobj.h"
 #include "main/mtypes.h"
--- ./src/mesa/swrast/s_texcombine.c~	2016-10-25 16:57:09.000000000 +0100
+++ ./src/mesa/swrast/s_texcombine.c	2016-10-25 17:20:20.297138713 +0100
@@ -24,6 +24,7 @@
  */
 
 
+#include <omp.h>
 #include "main/glheader.h"
 #include "main/context.h"
 #include "main/imports.h"
