--- ./include/SDL_opengl_glext.h~	2022-09-30 10:13:28.000000000 +0100
+++ ./include/SDL_opengl_glext.h	2022-09-30 10:14:51.289618713 +0100
@@ -758,8 +758,13 @@
 
 #ifndef GL_VERSION_1_5
 #define GL_VERSION_1_5 1
+#if __LP64__
 typedef khronos_ssize_t GLsizeiptr;
 typedef khronos_intptr_t GLintptr;
+#else
+typedef signed long GLsizeiptr;
+typedef signed long GLintptr;
+#endif
 #define GL_BUFFER_SIZE                    0x8764
 #define GL_BUFFER_USAGE                   0x8765
 #define GL_QUERY_COUNTER_BITS             0x8864
