--- libjava/Makefile.am~	2015-04-28 11:18:07.000000000 +0100
+++ libjava/Makefile.am	2015-04-29 22:26:27.407453410 +0100
@@ -44,7 +44,7 @@
 ##
 
 toolexeclib_LTLIBRARIES = libgcj.la libgij.la libgcj-tools.la
-toolexecmainlib_DATA = libgcj.spec
+toolexeclib_DATA = libgcj.spec
 
 ##
 ## These files form a closed dependency group at the lower
--- libjava/Makefile.in.orig	2015-06-21 09:55:19.155275814 +0100
+++ libjava/Makefile.in	2015-06-21 12:49:06.393765163 +0100
@@ -540,7 +540,7 @@
     n|no|NO) false;; \
     *) (install-info --version) >/dev/null 2>&1;; \
   esac
-DATA = $(dbexec_DATA) $(jar_DATA) $(toolexecmainlib_DATA)
+DATA = $(dbexec_DATA) $(jar_DATA) $(toolexeclib_DATA)
 RECURSIVE_CLEAN_TARGETS = mostlyclean-recursive clean-recursive	\
   distclean-recursive maintainer-clean-recursive
 AM_RECURSIVE_TARGETS = $(RECURSIVE_TARGETS:-recursive=) \
@@ -828,7 +828,7 @@
 libexecsubdir = $(libexecdir)/gcc/$(target_noncanonical)/$(gcc_version)
 toolexeclib_LTLIBRARIES = libgcj.la libgij.la libgcj-tools.la \
 	$(am__append_2) $(am__append_3) $(am__append_4)
-toolexecmainlib_DATA = libgcj.spec
+toolexeclib_DATA = libgcj.spec
 LOWER_PACKAGE_FILES_LO = \
 	$(propertyo_files) \
 	gnu-CORBA.lo \
@@ -10249,27 +10249,27 @@
 	@list='$(jar_DATA)'; test -n "$(jardir)" || list=; \
 	files=`for p in $$list; do echo $$p; done | sed -e 's|^.*/||'`; \
 	dir='$(DESTDIR)$(jardir)'; $(am__uninstall_files_from_dir)
-install-toolexecmainlibDATA: $(toolexecmainlib_DATA)
+install-toolexeclibDATA: $(toolexeclib_DATA)
 	@$(NORMAL_INSTALL)
-	@list='$(toolexecmainlib_DATA)'; test -n "$(toolexecmainlibdir)" || list=; \
+	@list='$(toolexeclib_DATA)'; test -n "$(toolexeclibdir)" || list=; \
 	if test -n "$$list"; then \
-	  echo " $(MKDIR_P) '$(DESTDIR)$(toolexecmainlibdir)'"; \
-	  $(MKDIR_P) "$(DESTDIR)$(toolexecmainlibdir)" || exit 1; \
+	  echo " $(MKDIR_P) '$(DESTDIR)$(toolexeclibdir)'"; \
+	  $(MKDIR_P) "$(DESTDIR)$(toolexeclibdir)" || exit 1; \
 	fi; \
 	for p in $$list; do \
 	  if test -f "$$p"; then d=; else d="$(srcdir)/"; fi; \
 	  echo "$$d$$p"; \
 	done | $(am__base_list) | \
 	while read files; do \
-	  echo " $(INSTALL_DATA) $$files '$(DESTDIR)$(toolexecmainlibdir)'"; \
-	  $(INSTALL_DATA) $$files "$(DESTDIR)$(toolexecmainlibdir)" || exit $$?; \
+	  echo " $(INSTALL_DATA) $$files '$(DESTDIR)$(toolexeclibdir)'"; \
+	  $(INSTALL_DATA) $$files "$(DESTDIR)$(toolexeclibdir)" || exit $$?; \
 	done
 
-uninstall-toolexecmainlibDATA:
+uninstall-toolexeclibDATA:
 	@$(NORMAL_UNINSTALL)
-	@list='$(toolexecmainlib_DATA)'; test -n "$(toolexecmainlibdir)" || list=; \
+	@list='$(toolexeclib_DATA)'; test -n "$(toolexeclibdir)" || list=; \
 	files=`for p in $$list; do echo $$p; done | sed -e 's|^.*/||'`; \
-	dir='$(DESTDIR)$(toolexecmainlibdir)'; $(am__uninstall_files_from_dir)
+	dir='$(DESTDIR)$(toolexeclibdir)'; $(am__uninstall_files_from_dir)
 
 # This directory's subdirectories are mostly independent; you can cd
 # into them and run `make' without going through this Makefile.
@@ -10546,7 +10546,7 @@
 install-exec-am: install-binPROGRAMS install-binSCRIPTS \
 	install-dbexecDATA install-dbexecLTLIBRARIES \
 	install-libexecsubPROGRAMS install-multi \
-	install-toolexeclibLTLIBRARIES install-toolexecmainlibDATA
+	install-toolexeclibLTLIBRARIES install-toolexeclibDATA
 	@$(NORMAL_INSTALL)
 	$(MAKE) $(AM_MAKEFLAGS) install-exec-hook
 install-html: install-html-recursive
@@ -10592,7 +10592,7 @@
 uninstall-am: uninstall-binPROGRAMS uninstall-binSCRIPTS \
 	uninstall-dbexecDATA uninstall-dbexecLTLIBRARIES \
 	uninstall-jarDATA uninstall-libexecsubPROGRAMS \
-	uninstall-toolexeclibLTLIBRARIES uninstall-toolexecmainlibDATA
+	uninstall-toolexeclibLTLIBRARIES uninstall-toolexeclibDATA
 
 .MAKE: $(RECURSIVE_CLEAN_TARGETS) $(RECURSIVE_TARGETS) all-multi \
 	clean-multi ctags-recursive distclean-multi install-am \
@@ -10616,7 +10616,7 @@
 	install-libexecsubPROGRAMS install-man install-multi \
 	install-pdf install-pdf-am install-ps install-ps-am \
 	install-strip install-toolexeclibLTLIBRARIES \
-	install-toolexecmainlibDATA installcheck installcheck-am \
+	install-toolexeclibDATA installcheck installcheck-am \
 	installdirs installdirs-am maintainer-clean \
 	maintainer-clean-generic maintainer-clean-multi mostlyclean \
 	mostlyclean-compile mostlyclean-generic mostlyclean-libtool \
@@ -10625,7 +10625,7 @@
 	uninstall-binSCRIPTS uninstall-dbexecDATA \
 	uninstall-dbexecLTLIBRARIES uninstall-jarDATA \
 	uninstall-libexecsubPROGRAMS uninstall-toolexeclibLTLIBRARIES \
-	uninstall-toolexecmainlibDATA
+	uninstall-toolexeclibDATA
 
 export gcc_version
 
