--- ./src/intel/intel_driver.c~	2016-09-13 07:25:11.000000000 +0100
+++ ./src/intel/intel_driver.c	2016-09-13 08:59:45.706785454 +0100
@@ -955,14 +955,15 @@
 
 #ifdef HAS_POOLED_EU
   /* BXT pooled eu, 3*6 to 2*9, like sub slice count is 2 */
-  unsigned int has_pooled_eu = 0;
-  if(!drm_intel_get_pooled_eu(driver->fd, &has_pooled_eu) && has_pooled_eu)
+  unsigned int has_pooled_eu = drm_intel_get_pooled_eu(driver->fd);
+
+  if (has_pooled_eu > 0)
     device->sub_slice_count = 2;
 
 #ifdef HAS_MIN_EU_IN_POOL
-  unsigned int min_eu;
   /* for fused down 2x6 devices, beignet don't support. */
-  if (has_pooled_eu && !drm_intel_get_min_eu_in_pool(driver->fd, &min_eu)) {
+  unsigned int min_eu = drm_intel_get_min_eu_in_pool(driver->fd);
+  if (min_eu <= 0) {
     assert(min_eu == 9); //don't support fuse down device.
   }
 #endif //HAS_MIN_EU_IN_POOL
