--- ./benchmark/CMakeLists.txt~	2014-04-30 18:19:34.000000000 +0100
+++ ./benchmark/CMakeLists.txt	2014-04-30 18:26:23.314810046 +0100
@@ -14,8 +14,8 @@
 
 ADD_LIBRARY(benchmarks SHARED ${ADDMATHFUNC} ${benchmark_sources})
 
-#TARGET_LINK_LIBRARIES(benchmarks cl m ${OPENGL_LIBRARIES} ${CMAKE_THREAD_LIBS_INIT})
-TARGET_LINK_LIBRARIES(benchmarks cl m)
+#TARGET_LINK_LIBRARIES(benchmarks OpenCL m ${OPENGL_LIBRARIES} ${CMAKE_THREAD_LIBS_INIT})
+TARGET_LINK_LIBRARIES(benchmarks OpenCL m)
 
 ADD_EXECUTABLE(benchmark_run benchmark_run.cpp)
 TARGET_LINK_LIBRARIES(benchmark_run benchmarks)
--- ./src/CMakeLists.txt.orig	2016-09-13 07:17:44.212816930 +0100
+++ ./src/CMakeLists.txt	2016-09-13 07:19:24.890382610 +0100
@@ -156,10 +156,11 @@
 SET(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wl,-Bsymbolic,--allow-shlib-undefined")
 
 link_directories (${LLVM_LIBRARY_DIR} ${DRM_LIBDIR} ${OPENGL_LIBDIR} ${EGL_LIBDIR})
-add_library(cl SHARED ${OPENCL_SRC})
-ADD_DEPENDENCIES(cl ${GIT_SHA1})
+add_library(OpenCL SHARED ${OPENCL_SRC})
+set_target_properties(OpenCL PROPERTIES VERSION 1.0.0 SOVERSION 1)
+ADD_DEPENDENCIES(OpenCL ${GIT_SHA1})
 target_link_libraries(
-                      cl
+                      OpenCL
                       rt
                       ${X11_LIBRARIES}
                       ${XEXT_LIBRARIES}
@@ -170,4 +171,4 @@
                       ${CMAKE_DL_LIBS}
                       ${OPENGL_LIBRARIES}
                       ${EGL_LIBRARIES})
-install (TARGETS cl LIBRARY DESTINATION ${BEIGNET_INSTALL_DIR})
+install (TARGETS OpenCL LIBRARY DESTINATION ${BEIGNET_INSTALL_DIR})
--- ./utests/CMakeLists.txt.orig	2016-09-13 07:17:44.212816930 +0100
+++ ./utests/CMakeLists.txt	2016-09-13 07:20:18.640865315 +0100
@@ -364,7 +364,7 @@
 ADD_LIBRARY(utests SHARED ${utests_sources})
 
 if (NOT_BUILD_STAND_ALONE_UTEST)
-  TARGET_LINK_LIBRARIES(utests cl m ${UTESTS_REQUIRED_GL_EGL_X11_LIB} ${CMAKE_THREAD_LIBS_INIT} ${UTESTS_REQUIRED_X11_LIB})
+  TARGET_LINK_LIBRARIES(utests OpenCL m ${UTESTS_REQUIRED_GL_EGL_X11_LIB} ${CMAKE_THREAD_LIBS_INIT} ${UTESTS_REQUIRED_X11_LIB})
 else()
   TARGET_LINK_LIBRARIES(utests ${OPENCL_LIBRARIES} m ${UTESTS_REQUIRED_GL_EGL_X11_LIB} ${CMAKE_THREAD_LIBS_INIT} ${UTESTS_REQUIRED_X11_LIB})
 endif()
