--- backend/src/CMakeLists.txt~	2013-02-28 12:12:45.000000000 +0000
+++ backend/src/CMakeLists.txt	2013-02-28 12:13:43.555682848 +0000
@@ -100,14 +100,17 @@
 target_link_libraries (gbe
   LLVMGenBackend
   LLVMTransformUtils
-  LLVMCore
-  LLVMAnalysis
   LLVMCodeGen
   LLVMScalarOpts
+  LLVMAnalysis
   LLVMSelectionDAG
-  LLVMSupport
   LLVMAsmParser
-  LLVMBitReader)
+  LLVMBitReader
+  LLVMTarget
+  LLVMCore
+  LLVMSupport
+  ${LLVM_LFLAGS}
+)
 
 install (TARGETS gbe LIBRARY DESTINATION lib)
 install (FILES backend/program.h DESTINATION include/gen)
