--- ./tools/clang/lib/Driver/Tools.cpp.orig	2013-12-14 08:07:38.171496314 +0000
+++ ./tools/clang/lib/Driver/Tools.cpp	2013-12-14 08:12:33.796203884 +0000
@@ -222,7 +222,7 @@
   // libprofile_rt.so. We used to use the -l:libprofile_rt.a syntax, but that is
   // not supported by old linkers.
   std::string ProfileRT =
-    std::string(TC.getDriver().Dir) + "/../lib/libprofile_rt.a";
+    "-l:llvm/libprofile_rt.a";
 
   CmdArgs.push_back(Args.MakeArgString(ProfileRT));
 }
@@ -5912,7 +5912,7 @@
   // forward.
   if (D.IsUsingLTO(Args)) {
     CmdArgs.push_back("-plugin");
-    std::string Plugin = ToolChain.getDriver().Dir + "/../lib/LLVMgold.so";
+    std::string Plugin = ToolChain.getDriver().Dir + "/..//@libdir@/llvm/LLVMgold.so";
     CmdArgs.push_back(Args.MakeArgString(Plugin));
 
     // Try to pass driver level flags relevant to LTO code generation down to
