--- ./tools/lldb/scripts/Python/finishSwigPythonLLDB.py~	2016-03-11 14:57:12.000000000 +0000
+++ ./tools/lldb/scripts/Python/finishSwigPythonLLDB.py	2016-03-11 17:24:04.255725154 +0000
@@ -719,8 +719,8 @@
     if bOk:
         bOk, strMsg = create_symlinks(vDictArgs, strFrameworkPythonDir)
 
-    if bOk:
-        bOk, strMsg = copy_six(vDictArgs, strFrameworkPythonDir)
+#    if bOk:
+#        bOk, strMsg = copy_six(vDictArgs, strFrameworkPythonDir)
 
     if bOk:
         bOk, strMsg = copy_lldbpy_file_to_lldb_pkg_dir(vDictArgs,
