--- ./libavcodec/vaapi_encode.c~	2017-12-28 16:08:51.000000000 +0000
+++ ./libavcodec/vaapi_encode.c	2017-12-28 18:47:22.120880058 +0000
@@ -1131,6 +1131,7 @@
 static av_cold int vaapi_encode_init_rate_control(AVCodecContext *avctx)
 {
     VAAPIEncodeContext *ctx = avctx->priv_data;
+    AVCPBProperties *cpb_props;
     int rc_bits_per_second;
     int rc_target_percentage;
     int rc_window_size;
@@ -1210,6 +1211,14 @@
         sizeof(ctx->fr_params);
 #endif
 
+    cpb_props = ff_add_cpb_side_data(avctx);
+    if (!cpb_props)
+        return AVERROR(ENOMEM);
+    cpb_props->max_bitrate = avctx->rc_max_rate;
+    cpb_props->min_bitrate = avctx->rc_min_rate;
+    cpb_props->avg_bitrate = rc_bits_per_second;
+    cpb_props->buffer_size = hrd_buffer_size;
+
     return 0;
 }
 
--- ./libavcodec/vaapi_encode.c~	2017-12-28 19:52:19.000000000 +0000
+++ ./libavcodec/vaapi_encode.c	2017-12-28 19:52:48.898319316 +0000
@@ -24,6 +24,7 @@
 #include "libavutil/log.h"
 #include "libavutil/pixdesc.h"
 
+#include "internal.h"
 #include "vaapi_encode.h"
 #include "avcodec.h"
 
