--- ./libavcodec/vaapi_encode_mpeg2.c~	2017-12-28 15:55:44.000000000 +0000
+++ ./libavcodec/vaapi_encode_mpeg2.c	2017-12-28 15:56:28.781804286 +0000
@@ -224,7 +224,7 @@
 
     if (avctx->sample_aspect_ratio.num != 0 &&
         avctx->sample_aspect_ratio.den != 0) {
-        AVRational dar = av_div_q(avctx->sample_aspect_ratio,
+        AVRational dar = av_div_q(av_inv_q(avctx->sample_aspect_ratio),
                                   (AVRational) { avctx->width, avctx->height });
 
         if (av_cmp_q(avctx->sample_aspect_ratio, (AVRational) { 1, 1 }) == 0) {
