--- ./libavcodec/vaapi_encode.c.orig	2018-10-04 13:00:56.850438199 +0000
+++ ./libavcodec/vaapi_encode.c	2018-10-04 13:02:27.611543483 +0000
@@ -24,6 +24,7 @@
 #include "libavutil/log.h"
 #include "libavutil/pixdesc.h"
 
+#include "internal.h"
 #include "vaapi_encode.h"
 #include "avcodec.h"
 
@@ -1209,6 +1210,7 @@
 static av_cold int vaapi_encode_init_rate_control(AVCodecContext *avctx)
 {
     VAAPIEncodeContext *ctx = avctx->priv_data;
+    AVCPBProperties *cpb_props;
     int64_t rc_bits_per_second;
     int     rc_target_percentage;
     int     rc_window_size;
@@ -1393,6 +1395,14 @@
                                   sizeof(ctx->fr_params));
 #endif
 
+    cpb_props = ff_add_cpb_side_data(avctx);
+    if (!cpb_props)
+        return AVERROR(ENOMEM);
+    cpb_props->max_bitrate = avctx->rc_max_rate;
+    cpb_props->min_bitrate = avctx->rc_min_rate;
+    cpb_props->avg_bitrate = rc_bits_per_second;
+    cpb_props->buffer_size = hrd_buffer_size;
+
     return 0;
 }
 
