--- configure.py~	2013-10-24 14:48:08.000000000 +0100
+++ configure.py	2013-10-24 14:48:32.119871871 +0100
@@ -96,6 +96,7 @@
        description = 'OPT $out')
 
 c_compiler_rule(b, "LLVM_TOOL_CXX", 'LLVM-CXX', 'clang++', llvm_cxxflags)
+c_compiler_rule(b, "LLVM_TOOL_LINK", 'LLVM-CXX', 'clang++', llvm_cxxflags)
 
 prepare_builtins = os.path.join('utils', 'prepare-builtins')
 b.build(os.path.join('utils', 'prepare-builtins.o'), "LLVM_TOOL_CXX",
