diff --git a/CMakeLists.txt b/CMakeLists.txt
index b9bd0df..22d51fb 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -50,17 +50,19 @@ if(MSVC)
   endif(NOT SHADERC_ENABLE_SHARED_CRT)
 endif(MSVC)
 
-# Configure subdirectories.
-# We depend on these for later projects, so they should come first.
-add_subdirectory(third_party)
+# Force cmake to link glslang before HLSL
+find_library(LIB_GLSLANG NAMES glslang)
+find_library(LIB_HLSL NAMES HLSL)
+add_library(glslang UNKNOWN IMPORTED)
+add_library(HLSL UNKNOWN IMPORTED)
+set_property(TARGET glslang PROPERTY IMPORTED_LOCATION "${LIB_GLSLANG}")
+set_property(TARGET HLSL PROPERTY IMPORTED_LOCATION "${LIB_HLSL}")
+
+add_dependencies(glslang HLSL)
 
+# Configure subdirectories.
 add_subdirectory(libshaderc_util)
 add_subdirectory(libshaderc)
 add_subdirectory(glslc)
 add_subdirectory(examples)
 
-add_custom_target(build-version
-  ${PYTHON_EXE}
-  ${CMAKE_CURRENT_SOURCE_DIR}/utils/update_build_version.py
-  ${shaderc_SOURCE_DIR} ${spirv-tools_SOURCE_DIR} ${glslang_SOURCE_DIR}
-  COMMENT "Update build-version.inc in the Shaderc build directory (if necessary).")
